File: problem-solving-js/arrayOfArrayProducts.js
function arrayOfArrayProducts(arr) {
  let product = [];
  
  if(arr.length == 0 || arr.length == 1) return product;
  
  for(let i = 0; i < arr.length; i++){
    let prod = 1;
    for( let j = 0; j < arr.length; j++){
      if(i != j){
       prod *= arr[j];
     }
    }
    product[i] = prod;
   }
   return product;
}

arrayOfArrayProducts([1,2,3,4])

File: problem-solving-rb/array_of_array_products.rb
def array_of_array_products(arr)
  product = []

  return product if arr.length == 0 || arr.length == 1

  arr.each_index do |i|
    prod = 1
    arr.each_index do |j|
      prod *= arr[j] unless i == j
    end
    product[i] = prod
  end

  product
end

result = array_of_array_products([1, 2, 3, 4])
puts result.inspect
